### PR TITLE
add blocklist/block api

### LINKED ISSUE #13706: Implement API to fetch details about blocks
API side of https://github.com/mozilla/addons-frontend/issues/9216

The API only needs to fetch details about a specific block, not list them. It needs to support fetching by add-on `guid` + version or legacy block ID.

### PR DESCRIPTION
**Pull request by eviljeff**
fixes #13706

### CODE REVIEW

File: src/addons.code-workspace [hunk 1]
```diff
@@ -0,0 +1,13 @@
+{
```
Comment (willdurand) [inline src/addons.code-workspace]: You probably want to remove this file (from the current patch).

Comment (eviljeff) [reply src/addons.code-workspace]: heh, whoops

**Comment by willdurand** [comment]
> Sorry, I forgot to ask the following question: in both the frontend and [server](https://github.com/mozilla/addons-server/issues/13706) issues, we mention a "version" but I don't see anything related to versions in this patch. What's the plan?

According to https://github.com/mozilla/addons-frontend/issues/9216#issuecomment-596680739, there is no need to specify a `version`.

File: src/olympia/blocklist/serializers.py [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+            'id', 'created', 'modified', 'guid', 'min_version', 'max_version',
+            'reason', 'url')
+
+    def to_representation(self, obj):
+        data = super().to_representation(obj)
+
+        if ('request' in self.context and
+                'wrap_outgoing_links' in self.context['request'].GET and
+                'url' in data):
+            data['url'] = get_outgoing_url(data['url'])
```
[the remark this answers is not in the document]
Comment (eviljeff) [reply src/olympia/blocklist/serializers.py]: we don't, but each serializer does something slightly different (AddonSerializer has to handle l10n, and hero shelves is already a mixin that deals with the CTA text also)

File: src/olympia/blocklist/tests/test_serializers.py [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+            'modified': block.modified.isoformat()[:-7] + 'Z',
+        }
+
+    def test_wrap_outgoing_links(self):
+        block = Block.objects.create(
+            guid='foo@baa',
+            min_version='45',
+            reason='something happened',
+            url='https://goo.gol',
+            updated_by=user_factory())
```
[the remark this answers is not in the document]
Comment (eviljeff) [reply src/olympia/blocklist/tests/test_serializers.py]: https://github.com/mozilla/addons-server/pull/13737/commits/5efe856b8108f473b6a12515aa357ba8043e49e3

**Pull request merged**